core/services/pipeline/common_eth.go

```diff
@@ -252,13 +252,18 @@ func convertToETHABIBytes(destType reflect.Type, srcVal reflect.Value, length in
 		return destVal.Interface(), nil
 
 	case reflect.Array:
-		if destType.Len() != length {
+		if destType.Kind() == reflect.Array && destType.Len() != length {
```

Comment:
`Len()` is only valid for `Array` `Kind`, and slices will be dynamically sized below.